components/net/http_loader.rs

```diff
@@ -411,12 +411,9 @@ fn obtain_response(
                     let chunk_requester = chunk_requester.clone();
                     let sender = sender.clone();
 
-                    devtools_bytes
-                        .upgrade()
-                        .expect("Couldn't upgrade devtools bytes weak.")
-                        .lock()
-                        .unwrap()
-                        .append(&mut bytes.clone());
+                    if let Some(devtools_bytes) = devtools_bytes.upgrade() {
```

Comment:
What was the problem we were hitting before this change?

Reply:
I think what happened was the main fetch would have exited already, and then the IPC route would execute. 
I think the only reason I'm using a `Weak` here, is so that I can unwrap the `Arc` below, where the bytes are actually sent in the message to devtools. 

If I were to use an `Arc` all over, I think we'd have to clone the bytes to move them into the message?

Reply:
I suspect cloning is necessary? We don't want to suppress devtools messages just because of timing differences in the fetch code.

Reply:
Yes, it's rather that if the fetch has already ended, and there are no more bytes that should be sent to devtools, then the arc will already have been dropped, and the IPC route will not be able to upgrade the weak. 

We do indeed clone into `devtools_bytes`, and I think only having one arc allows us to then move the contents of `devtools_bytes` into the devtools message, without an additional clone(not that this would matter much, just clarifying what I meant).

Reply:
I'll re-add an unwrap also and check exactly what triggered the panic that made me write this change...